### PR TITLE
Add a :verify_commit option

### PR DESCRIPTION
**Pull request by mohamedhafez**
When :verify_commit is set to true, after updating the mirror
`git verify-commit <revision>` will be run on the revision that is
about to be deployed to check if it has a valid signature. This
will allow a user to be certain that the code they are about to
deploy was signed by an authorized author, and that they are not
in danger of deploying malicious code in an attack scenario such as
https://github.blog/2012-03-04-public-key-security-vulnerability-and-mitigation/

discussed in https://github.com/capistrano/capistrano/issues/2074

### Short checklist

- [✔] Did you run `bundle exec rubocop -a` to fix linter issues?
- [✔] If relevant, did you create a test?
- [✔] Did you confirm that the RSpec tests pass?

### CODE REVIEW

**Comment by leehambley** [comment]
Thanks for the contribution @mohamedhafez I need to make a bit of time to just check naming consistency, and maybe just re-read our docs and make sure everything all continues to make sense but first impression looks really good.

I also noticed you didn't include an integration test, just units, which I think is fine, but I need to just invest a few more :brain: cycles to make sure I'm comfortable with that.

I'll get back to you really soon, thanks again!

**Comment by leehambley** [comment]
@mattbrictson I just realized I left @mohamedhafez hanging here. Did you have a strong opinion about merging this, or not?

**Comment by mattbrictson** [comment]
> @mattbrictson I just realized I left @mohamedhafez hanging here. Did you have a strong opinion about merging this, or not?

This a really nice PR with docs and tests, so from that perspective it looks good. However I know nothing about the `git verify-commit` command so I can't really speak to the usefulness of this feature or how we will maintain it. I'll leave it up to you. 🙂

File: docs/documentation/getting-started/configuration/index.markdown
Comment (leehambley) [inline docs/documentation/getting-started/configuration/index.markdown:161]: Hi @mohamedhafez so I'm more than happy to merge this if we change `:verify_commit` to `:git_verify_commit` in accordance with the existing `git` specific variables documented in this file (also at https://capistranorb.com/documentation/getting-started/configuration/)

That was the only issue I noticed, just consistency. I'm not sure other version control systems have a general purpose commit verification system, at least I searched a bit and didn't find anything, so I think we just focus on Git, for now?

**Review by leehambley — CHANGES_REQUESTED** [review]

File: docs/documentation/getting-started/configuration/index.markdown
Comment (mohamedhafez) [reply docs/documentation/getting-started/configuration/index.markdown]: @leehambley done! made the changes and its passing all tests

**Comment by LevonAr** [comment]
It looks like it went through. Nice one Mo!

**Pull request merged**
